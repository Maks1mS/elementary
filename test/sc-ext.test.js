import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import elementary from '../src';

const Button = styled.button`
  color: red;
`;

test('sc setup works', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', 'red');
});

const Box = elementary('div')(
  { effects: { hover: ':hover' }, styles: ['color'], aliases: true, mixins: true },
  { color: 'indigo' },
);

const Empty = elementary('div')({ styles: [] });

describe('Elementary integration to SC extend', () => {
  test('extend with inline', () => {
    const tree = renderer.create(<Box hoverC="red" c="blue" />).toJSON();
    expect(tree).toHaveStyleRule('color', 'blue');
    expect(tree).toHaveStyleRule('color', 'red', {
      modifier: ':hover',
    });
  });
  test('extend without inline', () => {
    const tree = renderer.create(<Box hoverC="red" />).toJSON();
    expect(tree).toHaveStyleRule('color', 'indigo');
    expect(tree).toHaveStyleRule('color', 'red', {
      modifier: ':hover',
    });
  });
  test('render without props', () => {
    const tree = renderer.create(<Empty />).toJSON();
    expect(tree).toBeDefined();
  });
});
