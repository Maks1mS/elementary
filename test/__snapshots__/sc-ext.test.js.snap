// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sc setup works 1`] = `
.c0 {
  color: red;
}

<button
  className="c0"
/>
`;
